fontir/src/ir.rs

```diff
@@ -99,18 +99,73 @@ impl Features {
 
 /// A variable definition of a single glyph.
 ///
-/// Defined in at least one position. If defined in
-/// many, presumed to vary continuously between positions and required
-/// to have variation compatible structure.
+/// Guarrantees at least one definition. Currently that must be at
+/// the default location. In theory that limitation is unnecessary,
+/// a variable glyph that covers the default location could compile.
+///
+/// If defined in many locations, presumed to vary continuously
+/// between positions and required to have variation compatible structure.
 #[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
 #[serde(from = "GlyphSerdeRepr", into = "GlyphSerdeRepr")]
 pub struct Glyph {
     pub name: GlyphName,
     pub codepoints: HashSet<u32>, // single unicodes that each point to this glyph. Typically 0 or 1.
-    pub sources: HashMap<NormalizedLocation, GlyphInstance>,
+    default_location: NormalizedLocation,
+    sources: HashMap<NormalizedLocation, GlyphInstance>,
 }
 
 impl Glyph {
+    pub fn new(
+        name: GlyphName,
+        codepoints: HashSet<u32>,
+        sources: HashMap<NormalizedLocation, GlyphInstance>,
+    ) -> Result<Self, WorkError> {
+        if sources.is_empty() {
+            return Err(WorkError::InvalidSourceGlyph {
+                glyph_name: name,
+                message: "No sources".into(),
+            });
+        }
+        let defaults: Vec<_> = sources
+            .keys()
+            .filter(|loc| !loc.iter().any(|(_, c)| c.into_inner() != 0.0))
```

Comment:
some thoughts:

- we could impl `PartialEq<f32>` for the various 'coord' types
- we could add `ZERO` consts on these types
- we could ignore me and move on

Reply:
I think PartialEq and ZERO are good ideas but I would like to do it in a separate PR. Hm. Zero arguably makes most sense for normalized coords, it doesn't have special significance for the others.

Reply:
https://github.com/googlefonts/fontmake-rs/issues/170